### PR TITLE
Update Discover logo

### PR DESCRIPTION
**Pull request by jackieluc**
**Checklist**
- [x] All icons have a corresponding entry in `db/payment_icons.yml`
- [x] I have followed the icon guidelines detailed in the [CONTRIBUTING.md](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md) file
- [x] I have optimized the icon with [SVGO](https://jakearchibald.github.io/svgomg/)
- [x] I am confident that all icons are clear and easy to read/understand
- [x] I have provided a link to the brand icon’s brand guidelines whenever possible.

If this pull request is not adding new icons, you can remove this checklist.

---

Update the discover logo from old discover icon to new discover icon.

My concerns revolve around the fact that I was unable to correctly render the logo when I modify the `viewBox` property to meet the icon guidelines in [CONTRIBUTING.md](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md). eg. It currently is `viewBox="60 128.4 380 243.1"` instead of `viewBox="0 0 38 24"`. My impression is that there are some elements that depend on this viewBox, but it doesn't seem trivial to adjust. If this is seen as an issue, please advise me on how to address this.

I was unable to find brand guidelines, but here are some of their media assets:
- https://www.discoverglobalnetwork.com/en-us/business-resources/free-signage-logos
- https://www.discover.com/company/newsroom/media-downloads/

### CODE REVIEW

**Comment by larouxn** [comment]
👋  @jackieluc, thanks for the update. Can we fix up the border on the new icon? (match the previous icon)

>Whenever possible, the border should be black (hex color #000) with a 7% opacity (0.07).
> \- [docs](https://github.com/activemerchant/payment_icons/blob/master/CONTRIBUTING.md#appearance)

Screen Shot 2021-04-02 at 12 21 51

**Comment by francispelland** [comment]
Once the border is fixed, should be good to merge!

**Comment by jackieluc** [comment]
Thanks @larouxn. I've updated the border.

**Comment by larouxn** [comment]
👋 Hey @jackieluc, thanks for editing the SVG. That said, I think it still needs some more tweaking. We're hoping to have the rounded corners like all of the other icons. I tried to fix it up but fell short on cutting off the orange swoosh thing at the bottom right rounded border. Not sure how to accomplish that.
Screen Shot 2021-04-07 at 10 04 58

<details>
  <summary>SVG fix attempt markup</summary>

  ```SVG
<svg width="38" height="24" viewBox="0 0 38 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M0 4C0 1.79086 1.79086 0 4 0H34C36.2091 0 38 1.79086 38 4V20C38 22.2091 36.2091 24 34 24H4C1.79086 24 0 22.2091 0 20V4Z" fill="white"/>
<path d="M37 13C37 13 27.1103 19.9041 9 23H37V13Z" fill="#F48120"/>
<path d="M4 0.5H34C35.933 0.5 37.5 2.067 37.5 4V20C37.5 21.933 35.933 23.5 34 23.5H4C2.067 23.5 0.5 21.933 0.5 20V4C0.5 2.067 2.067 0.5 4 0.5Z" stroke="black" stroke-opacity="0.07"/>
<path d="M3.57 7.15999H2V12.66H3.57C4.4 12.66 5 12.46 5.53 12.03C6.16 11.51 6.53 10.73 6.53 9.91999C6.52 8.28999 5.31 7.15999 3.57 7.15999V7.15999ZM4.83 11.3C4.49 11.6 4.06 11.74 3.36 11.74H3.07V8.09999H3.36C4.05 8.09999 4.47 8.21999 4.83 8.53999C5.2 8.86999 5.42 9.37999 5.42 9.90999C5.42 10.44 5.2 10.97 4.83 11.3Z" fill="#231F20"/>
<path d="M7.01999 7.15999H8.08999V12.66H7.01999V7.15999Z" fill="#231F20"/>
<path d="M10.71 9.27002C10.07 9.03002 9.88 8.87002 9.88 8.58002C9.88 8.23002 10.22 7.97002 10.68 7.97002C11 7.97002 11.27 8.10002 11.54 8.42002L12.1 7.69002C11.64 7.29002 11.09 7.08002 10.48 7.08002C9.51 7.08002 8.76 7.76002 8.76 8.66002C8.76 9.42002 9.11 9.81002 10.11 10.17C10.53 10.32 10.74 10.42 10.85 10.48C11.06 10.62 11.17 10.82 11.17 11.05C11.17 11.5 10.82 11.83 10.34 11.83C9.82999 11.83 9.42 11.57 9.17 11.1L8.48 11.77C8.97 12.5 9.57 12.82 10.38 12.82C11.49 12.82 12.28 12.08 12.28 11.01C12.3 10.12 11.93 9.72002 10.71 9.27002V9.27002Z" fill="#231F20"/>
<path d="M12.63 9.91999C12.63 11.54 13.9 12.79 15.53 12.79C15.99 12.79 16.39 12.7 16.87 12.47V11.21C16.44 11.64 16.06 11.81 15.58 11.81C14.5 11.81 13.73 11.03 13.73 9.90999C13.73 8.84999 14.52 8.01999 15.53 8.01999C16.04 8.01999 16.43 8.19999 16.87 8.63999V7.37999C16.4 7.13999 16.01 7.03999 15.55 7.03999C13.94 7.03999 12.63 8.31999 12.63 9.91999V9.91999Z" fill="#231F20"/>
<path d="M25.39 10.86L23.92 7.15999H22.75L25.08 12.8H25.66L28.03 7.15999H26.87L25.39 10.86Z" fill="#231F20"/>
<path d="M28.52 12.66H31.56V11.73H29.59V10.25H31.49V9.31999H29.59V8.09999H31.56V7.15999H28.52V12.66Z" fill="#231F20"/>
<path d="M35.81 8.79C35.81 7.76 35.1 7.17 33.86 7.17H32.27V12.67H33.34V10.46H33.48L34.96 12.67H36.28L34.55 10.35C35.36 10.18 35.81 9.63 35.81 8.79ZM33.65 9.7H33.34V8.03H33.67C34.34 8.03 34.7 8.31 34.7 8.85C34.7 9.4 34.34 9.7 33.65 9.7Z" fill="#231F20"/>
<path d="M20.16 12.86C21.7782 12.86 23.09 11.5482 23.09 9.93C23.09 8.31181 21.7782 7 20.16 7C18.5418 7 17.23 8.31181 17.23 9.93C17.23 11.5482 18.5418 12.86 20.16 12.86Z" fill="url(#paint0_linear)"/>
<path opacity="0.65" d="M20.16 12.86C21.7782 12.86 23.09 11.5482 23.09 9.93C23.09 8.31181 21.7782 7 20.16 7C18.5418 7 17.23 8.31181 17.23 9.93C17.23 11.5482 18.5418 12.86 20.16 12.86Z" fill="url(#paint1_linear)"/>
<path d="M36.49 9.37C36.49 9.27 36.42 9.22 36.31 9.22H36.15V9.7H36.27V9.51L36.41 9.7H36.55L36.39 9.5C36.45 9.49 36.49 9.44 36.49 9.37V9.37ZM36.29 9.44H36.27V9.31H36.29C36.35 9.31 36.38 9.33 36.38 9.37C36.38 9.42 36.35 9.44 36.29 9.44V9.44Z" fill="#231F20"/>
<path d="M36.33 9.03999C36.1 9.03999 35.91 9.22999 35.91 9.45999C35.91 9.68999 36.1 9.87999 36.33 9.87999C36.56 9.87999 36.75 9.68999 36.75 9.45999C36.75 9.22999 36.56 9.03999 36.33 9.03999ZM36.33 9.80999C36.15 9.80999 35.99 9.65999 35.99 9.45999C35.99 9.26999 36.14 9.10999 36.33 9.10999C36.51 9.10999 36.66 9.26999 36.66 9.45999C36.66 9.64999 36.51 9.80999 36.33 9.80999Z" fill="#231F20"/>
<defs>
<linearGradient id="paint0_linear" x1="21.6575" y1="12.2747" x2="19.6321" y2="9.1035" gradientUnits="userSpaceOnUse">
<stop stop-color="#F89F20"/>
<stop offset="0.25" stop-color="#F79A20"/>
<stop offset="0.533" stop-color="#F68D20"/>
<stop offset="0.62" stop-color="#F58720"/>
<stop offset="0.723" stop-color="#F48120"/>
<stop offset="1" stop-color="#F37521"/>
</linearGradient>
<linearGradient id="paint1_linear" x1="21.3378" y1="12.232" x2="18.3776" y2="6.4462" gradientUnits="userSpaceOnUse">
<stop stop-color="#F58720"/>
<stop offset="0.359" stop-color="#E16F27"/>
<stop offset="0.703" stop-color="#D4602C"/>
<stop offset="0.982" stop-color="#D05B2E"/>
</linearGradient>
</defs>
</svg>
  ```
</details>

From [CONTRIBUTING.md](https://github.com/activemerchant/payment_icons/blob/c7a2610ca16251de7dceea90c7fbe301877ad9aa/CONTRIBUTING.md):

> Icons have a visible border to give them a consistent shape on pages whose backgrounds match that of the icon.
> - Whenever possible, the border should be black (hex color #000) with a 7% opacity (0.07).
> - The border width must be 1px (pixel) thick.
> - The border must have a 2px radius (outside stroke).

**Comment by jackieluc** [comment]
Thanks @larouxn for the attempt. I took your attempt and was able to round out the orange shape in Figma. Here's what it looks like now:

**Review by larouxn — APPROVED** [review]
LGTM 🎩 
Screen Shot 2021-04-08 at 13 56 28

File: app/assets/images/payment_icons/discover.svg [hunk 1]
```diff
@@ -1 +1 @@
...
... 36.4609 7.50593C36.4609 7.55593 36.4309 7.57593 36.3709 7.57593Z" fill="#231F20"/><path d="M36.4109 7.17592C36.1809 7.17592 35.9909 7.36592 35.9909 7.59592C35.9909 7.82592 36.1809 8.01592 36.4109 8.01592C36.6409 8.01592 36.8309 7.82592 36.8309 7.59592C36.8309 7.36592 36.6409 7.17592 36.4109 7.17592ZM36.4109 7.94592C36.2309 7.94592 36.0709 7.79592 36.0709 7.59592C36.0709 7.40592 36.2209 7.24592 36.4109 7.24592C36.5909 7.24592 36.7409 7.40592 36.7409 7.59592C36.7409 7.78592 36.5909 7.94592 36.4109 7.94592Z" fill="#231F20"/><defs><linearGradient id="pi-paint0_linear" x1="21.657" y1="12.275" x2="19.632" y2="9.104" gradientUnits="userSpaceOnUse"><stop stop-color="#F89F20"/><stop offset=".25" stop-color="#F79A20"/><stop offset=".533" stop-color="#F68D20"/><stop offset=".62" stop-color="#F58720"/><stop offset=".723" stop-color="#F48120"/><stop offset="1" stop-color="#F37521"/></linearGradient><linearGradient id="pi-paint1_linear" x1="21.338" y1="12.232" x2="18.378" y2="6.446" gradientUnits="userSpaceOnUse"><stop stop-color="#F58720"/><stop offset=".359" stop-color="#E16F27"/><stop offset=".703" stop-color="#D4602C"/><stop offset=".982" stop-color="#D05B2E"/></linearGradient></defs></svg>
```
Comment (tauthomas01) [inline app/assets/images/payment_icons/discover.svg:1]: There seems to be a border radius difference. I think it has to do with the border settings in the illustration tool where you can set between "inside" or "outside" options.

[video demonstration](https://screenshot.click/08-08-blrbh-ne5mo.mp4)

**Review by tauthomas01 — CHANGES_REQUESTED** [review]

File: app/assets/images/payment_icons/discover.svg (hunk 1 above)
Comment (jackieluc) [reply app/assets/images/payment_icons/discover.svg]: Thanks @tauthomas01. Here's the latest iteration with an update to the border radius. I took the SVG elements from the Visa card to update this in Figma and re-added the attributes from the guidelines. Anything else to consider that I may be missing?

**Review by tauthomas01 — APPROVED** [review]
Looks good, thank you for the adjustments

**Comment by jackieluc** [comment]
Thank you @tauthomas01, when will a new version of the gem be released?

**Pull request merged**